pkgs/by-name/sw/sword/package.nix

```diff
@@ -17,12 +21,28 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-QkCc894vrxEIUj4sWsB0XSH57SpceO2HjuncwwNCa4o=";
   };
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [
-    icu
-    clucene_core
-    curl
-  ];
+  nativeBuildInputs =
+    [
+      pkg-config
+    ]
+    ++ (lib.optionals stdenv.hostPlatform.isWindows [
+      autoreconfHook # The Windows patch modifies autotools files
+      (curl.override { pslSupport = false; }).dev # Provides curl-config
```

Comment:
Since you're an upstream dev I'll say that you shouldn't use `-config` files, instead use pkg-config when possible, you're checking for `--protocols` which are available in the pc file as `supported_protocols`

Reply:
So you mean things like `curl-config` shouldn't be used by SWORD's autotools scripts?

I don't believe we're checking it for `--protocols` at all. Only for the library's existence. This override to disable `pslSupport` is only because cross-compiling the curl package for MinGW is broken due to its dependency on libpsl which depends on python3. That package currently fails because it's pulling Windows specific patches and passing them incorrectly to the patches array. Although I've fixed that logic locally, the patch set it pulls is for Python 3.11 instead of 3.12 and won't apply cleanly. Until that patchset (from Fedora's MinGW team) is updated or python3 is dropped from MinGW libpsl builds, we're stuck with this override.

The only officially supported protocols for SWORD usage of libcurl is FTP, HTTP, and HTTPS, but SWORD just constructs the URL blindly and passes it to libcurl and if it supports the protocol, so be it.

Reply:
> So you mean things like `curl-config` shouldn't be used by SWORD's autotools scripts?
> 
> I don't believe we're checking it for `--protocols` at all. Only for the library's existence. This override to disable `pslSupport` is only because cross-compiling the curl package for MinGW is broken due to its dependency on libpsl which depends on python3. That package currently fails because it's pulling Windows specific patches and passing them incorrectly to the patches array. Although I've fixed that logic locally, the patch set it pulls is for Python 3.11 instead of 3.12 and won't apply cleanly. Until that patchset (from Fedora's MinGW team) is updated or python3 is dropped from MinGW libpsl builds, we're stuck with this override.
> 
> The only officially supported protocols for SWORD usage of libcurl is FTP, HTTP, and HTTPS, but SWORD just constructs the URL blindly and passes it to libcurl and if it supports the protocol, so be it.

https://github.com/bibletime/crosswire-sword-mirror/blob/ec86f2bf9eb84f4a938637e07e2b81a769118c58/CMakeLists.txt#L214 https://github.com/bibletime/crosswire-sword-mirror/blob/ec86f2bf9eb84f4a938637e07e2b81a769118c58/configure.ac#L296 here's the `--protocols`

For autotools you should add
`"ac_cv_prog_CURL_CONFIG=${lib.getDev (curl.override { pslSupport = false; })}/bin/curl-config"` to `configureFlags`

The reason for not adding to `nativeBuildInputs` is cleanliness

The `-config` scripts are a leftover from the time when pkg-config(created in 2000) did not yet exist.

Reply:
> [bibletime/crosswire-sword-mirror@`ec86f2b`/CMakeLists.txt#L214](https://github.com/bibletime/crosswire-sword-mirror/blob/ec86f2bf9eb84f4a938637e07e2b81a769118c58/CMakeLists.txt#L214) [bibletime/crosswire-sword-mirror@`ec86f2b`/configure.ac#L296](https://github.com/bibletime/crosswire-sword-mirror/blob/ec86f2bf9eb84f4a938637e07e2b81a769118c58/configure.ac#L296) here's the `--protocols`

Ah yes, I forgot we had put in the SFTP conditional before that release.

> 
> For autotools you should add `"ac_cv_prog_CURL_CONFIG=${lib.getDev (curl.override { pslSupport = false; })}/bin/curl-config"` to `configureFlags`
>

This doesn't work. CURL remains undiscovered by autotools.
 
> The reason for not adding to `nativeBuildInputs` is cleanliness
> 
> The `-config` scripts are a leftover from the time when pkg-config(created in 2000) did not yet exist.

Yes, and the SWORD library also dates from that before time. I'm not going to be able to convince the lead developer to forgo use of those curl-config and icu-config in favor of pkg-config at this point. The code is still hosted in Subversion and fixes are sent as patches to a mailing list. Updating to the latest and greatest is not a priority for the project. Stability and extra small library footprint is the main concern.

Reply:
I know, no one wants to touch autotools.

Reply:
Oh, he updates the autotools stuff regularly. But just not for things he would deem unnecessary. Adding new features, or an optional library is fine. But those config scripts continue to be maintained for 20+ years, so I doubt he would see a benefit to moving away from them until they're gone.